lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1203,10 +1203,17 @@ def _build_files_manifest_walk(b_collection_path, namespace, name, ignore_patter
 
     manifest = _make_manifest()
 
+    def _discover_relative_base_directory(b_path: bytes, b_top_level_dir: bytes) -> bytes:
+        if b_path == b_top_level_dir:
+            return b''
+        common_prefix = os.path.commonpath((b_top_level_dir, b_path))
+        b_rel_base_dir = os.path.relpath(b_path, common_prefix)
+        return b_rel_base_dir.lstrip(os.path.sep.encode())
+
     def _walk(b_path, b_top_level_dir):
```

Comment:
Now that I realize that this change is trying to patch a private API for external use, it feels like a better fix would be introducing input validation instead.
Any input normalization must be done by the caller and our code already does so by calling https://github.com/ansible/ansible/blob/3b608f9/lib/ansible/cli/galaxy.py#L1046.